hpat/datatypes/hpat_pandas_series_functions.py

```diff
@@ -2512,15 +2512,274 @@ def hpat_pandas_series_median_impl(self, axis=None, skipna=True, level=None, num
     return hpat_pandas_series_median_impl
 
 
+@overload_method(SeriesType, 'argsort')
+def hpat_pandas_series_argsort(self, axis=0, kind='quicksort', order=None):
+    """
+    Pandas Series method :meth:`pandas.Series.argsort` implementation.
+
+    .. only:: developer
+
+       Test: python -m -k hpat.runtests hpat.tests.test_series.TestSeries.test_series_argsort*
+
+    Parameters
+    -----------
+    self: :class:`pandas.Series`
+        input series
+    axis: :obj:`int`
+        Has no effect but is accepted for compatibility with numpy.
+        *unsupported*
+    kind: :obj:'str', {'mergesort', 'quicksort', 'heapsort'}, default: 'quicksort'
+        Choice of sorting algorithm. See np.sort for more information. 'mergesort' is the only stable algorithm
+        *uses python func - sorted() for str and numpy func - sort() for num*
+        *unsupported*
+    order: :obj:`str` or  :obj:`list of str`, default: None
+        Has no effect but is accepted for compatibility with numpy.
+        *unsupported*
+
+    Returns
+    -------
+    :obj:`pandas.Series`
+         returns: Positions of values within the sort order with -1 indicating nan values.
+    """
+
+    _func_name = 'Method argsort().'
+
+    if not isinstance(self, SeriesType):
+        raise TypingError('{} The object must be a pandas.series. Given: {}'.format(_func_name, self))
+
+    if not isinstance(self.data.dtype, types.Number):
+        raise TypingError('{} Currently function supports only numeric values. Given data type: {}'.format(_func_name,
+                                                                                                           self.data.dtype))
+
+    if not (isinstance(axis, types.Omitted) or isinstance(axis, types.Integer) or axis == 0):
+        raise TypingError('{} Unsupported parameters. Given axis: {}'.format(_func_name, axis))
+
+    if not isinstance(self.index, types.NoneType):
+        def hpat_pandas_series_argsort_impl(self, axis=0, kind='quicksort', order=None):
```

Comment:
All these branches looks big. I think this is a good practice to insert a comments inside such branch with short explanations and test name for this branch.
First question will be asked here is "what differences between these branches?"

Reply:
also, please don't use same name of functions. it looks like you use `hpat_pandas_series_argsort_impl` for any branch